easybuild/easyblocks/c/code_saturne.py

```diff
@@ -0,0 +1,167 @@
+
+        self.log.info("Running ./sbin/bootstrap ...")
+        cmd = './sbin/bootstrap'
+        run_cmd(cmd, log_all=True, simple=True, log_output=True)
+
+        cmd = ' '.join([
+            self.cfg['preconfigopts'],
+            './configure',
+            '--prefix=' + self.installdir,
+            '--without-modules',
+            self.cfg['configopts'],
+            ])
+
+        if self.cfg['debug']:
+            cmd = ' '.join([cmd, '--enable-debug'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def build_step(self):
+        """ Build step for Code_Saturne."""
+
+        paracmd = ''
+        if self.cfg['parallel']:
+            paracmd = "-j %s" % self.cfg['parallel']
+
+        cmd = "%s make %s %s" % (self.cfg['prebuildopts'],
+                                 paracmd, self.cfg['buildopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def install_step(self):
+        """ Install step for Code_Saturne."""
+
+        cmd = "%s make install %s" % (self.cfg['preinstallopts'],
+                                      self.cfg['installopts'])
+
+        (out, _) = run_cmd(cmd, log_all=True, simple=False, log_output=True)
+
+        return out
+
+    def post_install_step(self):
+        """Custom post install step for Code_Saturne."""
+        super(EB_Code_underscore_Saturne, self).post_install_step()
+
+        # if we are using a SLURM system, we need to change the defaults;
+        # so, modify the template and rename it to "etc/code_saturne.cfg"
+        if self.cfg['slurm']:
+
+            self.log.info("Running the post-install SLURM step ...")
+
+            target_path = os.path.join(self.installdir, 'etc/code_saturne.cfg')
+            from_path = target_path + '.template'
+
+            apply_regex_substitutions(
+                from_path,
+                [(r"# batch =", r"batch = SLURM")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec = mpiexec", r"mpiexec = srun")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec_n = ' -n '", r"mpiexec_n = ' -n '")]
+                )
+            apply_regex_substitutions(
+                from_path,
+                [(r"# mpiexec_n_per_node =",
+                  r"mpiexec_n_per_node = ' --ntasks-per-node '")]
+                )
+
+            os.rename(from_path, target_path)
+
+    def sanity_check_step(self):
+        """Custom sanity check step for Code_Saturne."""
+
+        shlib_ext = get_shared_lib_ext()
+
+        custom_paths = {
+            'files': ['bin/code_saturne', 'lib/libsaturne.%s' % shlib_ext],
+            'dirs': ['bin', 'lib', 'libexec', 'include', 'etc'],
+        }
+
+        super(EB_Code_underscore_Saturne, self).sanity_check_step(
+            custom_paths=custom_paths)
```

Comment:
No need to wrap at all here?

```python
super(EB_Code_underscore_Saturne, self).sanity_check_step(custom_paths=custom_paths)

Reply:
Good to know. I somehow missed the note on your style page.